runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -159,16 +160,31 @@ protected boolean discoverAddressSpace(final UUID streamId,
                 queue.addAll(streamAddressSpace.copyAddressesToSet(maxGlobal));
 
                 long trimMark = streamAddressSpace.getTrimMark();
+
+                // In case we are dealing with a stream that does not have the checkpoint
+                // capability, check to see if we are trying to access an address that has been
+                // previously trimmed.
+                if (!isCheckpointCapable()
+                        && Address.isAddress(trimMark)
+                        && trimMark >= stopAddress) {
```

Comment:
Since stop address is exclusive it should be >, right?

Reply:
OK great. I was not sure about that one.

Reply:
Done.